src/dawn/SIR/SIR.h

```diff
@@ -159,9 +159,10 @@ struct StencilFunctionArg {
 /// @ingroup sir
 struct Field : public StencilFunctionArg {
   Field(const std::string& name, SourceLocation loc = SourceLocation())
-      : StencilFunctionArg{name, AK_Field, loc}, IsTemporary(false) {}
+      : StencilFunctionArg{name, AK_Field, loc}, IsTemporary(false), fieldDimensions({{0, 0, 0}}) {}
```

Comment:
just `fieldDimensions{0,0,0} ` ?

Reply:
we are actually using the double brackets pretty consistently for ArrayXi in out codes, examples [here](https://github.com/MeteoSwiss-APN/dawn/blob/26392782936f260876cf95b10e870a9df30a6459/src/dawn/Optimizer/AccessComputation.cpp#L150) or [here](https://github.com/MeteoSwiss-APN/dawn/blob/26392782936f260876cf95b10e870a9df30a6459/src/dawn/SIR/SIRSerializer.cpp#L763) amongst many others. 
I stuck to that style.
If you don't see a reason to do so, we can consider changing to single-bracket style?

Reply:
I was not aware of that, but what is the C++ reason behind that? 
Also here we are using triple brackets `({{`

Reply:
the round brackets are from the initializer-list, but the Array-Ctor has the 2 as anywhere else.

It is because c++11 does not let you write  `std::array<int, 3> foo{1,2,3}` but you are required to write `std::array<int, 3> foo{{1,2,3}}` whereas you are allowed to write `std::array<int, 3> foo = {1,2,3}`.
I think that  to keep the look of it more consistent, we went with with double-brackets everywhere